pkgs/crypto/benchmark/benchmark.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:typed_data';
+
+import 'package:convert/convert.dart';
+import 'package:crypto/crypto.dart';
+
+void main(List<String> args) {
+  Hash? function;
+
+  void setFunction(Hash newFunction, String message) {
+    if (function != null) throw ArgumentError('Hash function already set.');
+    function = newFunction;
+    print('Using hash function $message');
+  }
+
+  for (var arg in args) {
+    if (arg == 'md5') {
+      setFunction(md5, 'md5');
+    } else if (arg == 'sha1') {
+      setFunction(sha1, 'sha1');
+    } else if (arg == 'sha256') {
+      setFunction(sha256, 'sha256');
+    } else if (arg == 'sha224') {
+      setFunction(sha224, 'sha224');
+    } else if (arg == 'sha384') {
+      setFunction(sha384, 'sha384');
+    } else if (arg == 'sha512') {
+      setFunction(sha512, 'sha512');
+    } else if (arg == 'sha512224') {
+      setFunction(sha512224, 'sha512/224');
+    } else if (arg == 'sha512256') {
+      setFunction(sha512256, 'sha512/256');
+    } else {
+      throw ArgumentError('Unknown argument: $arg');
```

Comment:
Print an error on `stderr` and exit. This is a command line usage mistake, not a Dart argument error.